### PR TITLE
[Tech Debt] Nuke Encoded{Signature,PublicKey}

### PR DESCRIPTION
**Pull request by DieracDelta**
Closes issue: #963 

### This PR: 
This PR removes the EncodedSignature and EncodedPublicKey types.

### Key places to review: 
- the sign implementation for bn256
- genesis_proposer_pk

<!-- Complete the following items before creating this PR
* Are the proper people tagged to review it? Not yet, it's a draft
* Have you linked an issue to this PR?

### CODE REVIEW

File: crates/task-impls/src/da.rs [hunk 1]
```diff
@@ -260,8 +263,15 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
...
-                let signature =
-                    TYPES::SignatureKey::sign(&self.private_key, &encoded_transactions_hash);
+                let Ok(signature) =
+                    TYPES::SignatureKey::sign(&self.private_key, &encoded_transactions_hash)
+                else {
+                    // TODO is this correct?
+                    // Should we be doing more?
+                    error!("Failed to sign block payload!");
+                    return None;
+                };
```
Comment (dailinsubjam) [inline crates/task-impls/src/da.rs:273]: Looks fine to me. I think it's hard to fail in signing sth, unless it's a very extreme corner case.

File: crates/testing/tests/view_sync_task.rs [hunk 2]
```diff
@@ -34,7 +34,8 @@ async fn test_view_sync_task() {
         <TestTypes as hotshot_types::traits::node_implementation::NodeType>::Time::new(5),
         hotshot_types::traits::consensus_api::ConsensusSharedApi::public_key(&api),
         hotshot_types::traits::consensus_api::ConsensusSharedApi::private_key(&api),
-    );
+    )
+    .unwrap();
```
Comment (dailinsubjam) [inline crates/testing/tests/view_sync_task.rs:38]: Do we want to deal with this `unwarp()` safely by taking failed unwrap() into account?

**Review by dailinsubjam — COMMENTED** [review]
Looks good! Very glad to see these files getting cleaner.

File: crates/hotshot-signature-key/src/bn254/bn254_pub.rs [hunk 3]
```diff
@@ -51,47 +52,28 @@ impl SignatureKey for BLSPubKey {
...
-                // This is the validation for QC partial signature before append().
-                let generic_msg: &GenericArray<u8, U32> = GenericArray::from_slice(data);
-                BLSOverBN254CurveSignatureScheme::verify(&(), &ver_key, generic_msg, &s).is_ok()
-            }
-            Err(_) => false,
-        }
+
+        // This is the validation for QC partial signature before append().
+        let generic_msg: &GenericArray<u8, U32> = GenericArray::from_slice(data);
+        BLSOverBN254CurveSignatureScheme::verify(&(), &ver_key, generic_msg, signature).is_ok()
```
[the remark this answers is not in the document]
Comment (dailinsubjam) [reply crates/hotshot-signature-key/src/bn254/bn254_pub.rs:63]: Followup: Discussed 1:1, we'll keep it as it is now, and discuss the probability of " directly tie the jf trait in as a replacement for signaturekey" on the next pair programming.

**Review by rob-maron — DISMISSED** [review]
Love it !

**Review by dailinsubjam — DISMISSED** [review]

**Review by dailinsubjam — APPROVED** [review]
